File: 189-RotateArray/189-RotateArray.java
// Last updated: 28/06/2026, 01:03:55
class Solution {
    public void rotate(int[] nums, int k) {
        int n=nums.length;
        k=k%n;
        rotate(0,n-1,nums);
        rotate(0,k-1,nums);
        rotate(k,n-1,nums);
    }
    void rotate(int l,int r,int[] nums){
        while(l<=r){
            int temp=nums[l];
            nums[l]=nums[r];
            nums[r]=temp;
            l++;
            r--;
        }
    }
}

File: 189-RotateArray/189-RotateArray.cpp
// Last updated: 28/06/2026, 01:04:01
class Solution {
public:
    void reverse1(vector<int> & nums,int s,int e){
        while(s<=e)
        {
            int temp=nums[s];
            nums[s]=nums[e];
            nums[e]=temp;
            s++;
            e--;
        }
    }
    void rotate(vector<int>& nums, int k) {
        int n=nums.size();
         k = k%n;
        if(n==0 )
            return;
       
        reverse1(nums,n-k,n-1);
        reverse1(nums,0,n-k-1);
        reverse1(nums,0,n-1);
    }
};